robot/client/client.go

```diff
@@ -345,7 +345,7 @@ func (rc *RobotClient) connectWithLock(ctx context.Context) error {
 	if err := rc.conn.Close(); err != nil {
 		return err
 	}
-	conn, err := grpc.Dial(ctx, rc.address, rc.logger, rc.dialOptions...)
+	conn, err := grpc.Dial(ctx, rc.address, logging.GetOrNewLogger("rdk.network_outgoing"), rc.dialOptions...)
```

Comment:
This is a tough one, and I think it's slightly incorrect. You're making the gRPC dial logger `rdk.network_outgoing` for all Go SDK clients with this change. While the RDK uses Go SDK clients to connect to remotes, we don't want to mess with the Go SDK for people that use it as an actual SDK. I think you'll want to change [this](https://github.com/viamrobotics/rdk/blob/main/robot/impl/local_robot.go#L1129) logger instead.

Reply:
ah yeah that makes sense, thanks for calling out

Reply:
I decided to not go ahead with that change and instead change `logging.GetOrNewLogger("rdk.networking")` to `rc.logger.Sublogger("networking")` so we still get the `networking` suffix (and so will SDK users, but I think this is ok). The line you linked would only change connection logging for remotes, where I think it that case it would be valuable to still keep the resource name as the logger name (as well as adding the `networking` suffix).